src/AnalyticsEventReporter.ts

```diff
@@ -34,10 +34,27 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
   with(payload: EventPayload): AnalyticsEventService {
     const currentPayload =
       this.payload === undefined ? payload : merge(this.payload, payload);
+    if (this.config.debug) {
+      console.log('[DEBUG] AnalyticsConfig Object:\n', this.config);
+      console.log(
+        '[DEBUG] Payload following call to `with`:\n',
+        currentPayload
+      );
+    }
     return new AnalyticsEventReporter(this.config, currentPayload);
   }
 
   public async report(newPayload?: EventPayload): Promise<string> {
+    if (this.config.debug) {
+      console.log(
+        '[DEBUG] Merging the following payloads:\n' +
+          '\nOriginal Payload from call to `with`:\n' +
+          `\n${this.payload ? JSON.stringify(this.payload) : '{}'}\n` +
+          '\nNew Payload from call to `report:`\n' +
```

Comment:
Thoughts on simple helper for formatting debug messages? Mainly to clean up the readability of this report function + file.

Also helps so that all debug messages have consistent formatting!